core/src/main/java/com/rudderstack/android/sdk/core/FlushUtils.java

```diff
@@ -0,0 +1,244 @@
+package com.rudderstack.android.sdk.core;
+
+import android.text.TextUtils;
+
+import androidx.annotation.Nullable;
+
+import com.rudderstack.android.sdk.core.util.Utils;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Standard flush related calls
+ */
+class FlushUtils {
+    private FlushUtils() {
+
+    }
+
+    private static final Object FLUSH_LOCK = new Object();
+    private static final Object DB_LOCK = new Object();
+
+    /**
+     * Should not be called from main thread.
+     *
+     * @param areFactoriesInitialized
+     * @param integrationOperationsMap
+     * @param messageIds
+     * @param messages
+     * @param flushQueSize
+     * @param dataPlaneUrl
+     * @param dbManager
+     * @param authHeaderString
+     * @param anonymousIdHeaderString
+     * @return
+     */
+    static boolean flush(boolean areFactoriesInitialized, @Nullable Map<String, RudderIntegration<?>> integrationOperationsMap,
+                         List<Integer> messageIds, List<String> messages, int flushQueSize, String dataPlaneUrl,
+                         DBPersistentManager dbManager,
+                         String authHeaderString, String anonymousIdHeaderString) {
+
+        if (areFactoriesInitialized && integrationOperationsMap != null) {
+            RudderLogger.logDebug("EventRepository: flush native SDKs");
+            for (String key : integrationOperationsMap.keySet()) {
+                RudderLogger.logDebug(String.format(Locale.US, "EventRepository: flush for %s", key));
+                RudderIntegration<?> integration = integrationOperationsMap.get(key);
+                if (integration != null) {
+                    integration.flush();
+                }
+            }
+        }
+        Utils.NetworkResponses networkResponse;
+
+        messageIds.clear();
+        messages.clear();
+        RudderLogger.logDebug("EventRepository: flush: Fetching events to flush to server");
+        synchronized (DB_LOCK) {
+            dbManager.fetchAllEventsFromDB(messageIds, messages);
```

Comment:
Instead of sending the lists as arguments, isn't returning these values better? 
Shouldn't fetch type functions have return values.

Reply:
Actually this fetch api is supposed to send both messageIds list and Messages. 
This approach is followed in case of matrix operations, where empty list/array references are sent that are filled by the operation.
The better alternative in this case could have been a POJO/Map being returned with the values (standard practice for ORM). But since this is a legacy method, I didn't change it as it might break some other implementation.

Reply:
got it